FirebaseAuth/Sources/Public/FirebaseAuth/FIROAuthProvider.h

```diff
@@ -114,6 +114,22 @@ NS_SWIFT_NAME(OAuthProvider)
                                          IDToken:(NSString *)IDToken
                                         rawNonce:(nullable NSString *)rawNonce;
 
+/** @fn appleCredentialWithIDToken:rawNonce:fullName:
+ *  @brief Creates an `AuthCredential` for the Sign in with Apple OAuth 2 provider identified by ID
+ * token, raw nonce, and full name. This method is specific to the Sign in with Apple OAuth 2
+ * provider as this provider requires the full name to be passed explicitly.
+ *
+ *  @param IDToken The IDToken associated with the Sign in with Apple Auth credential being created.
+ *  @param rawNonce The raw nonce associated with the Sign in with Apple Auth credential being
+ * created.
+ *  @param fullName The full name associated with the Sign in with Apple Auth credential being
+ * created.
+ *  @return An `AuthCredential`.
+ */
++ (FIROAuthCredential *)appleCredentialWithIDToken:(NSString *)IDToken
+                                          rawNonce:(nullable NSString *)rawNonce
+                                          fullName:(nullable NSPersonNameComponents *)fullName;
```

Comment:
Why does `fullName` need to be nullable?

Reply:
So given that calling this method with a nil `fullName` is the same as calling `credentialWithProviderID` with the `apple.com` provider ID, `fullName` doesn't need to be nullable. However, given that this is an optional field, it seems more correct to make this nullable.